waiter/src/waiter/scheduler.clj

```diff
@@ -247,7 +252,19 @@
     (->> port-index dec (nth extra-ports))
     port))
 
-(defn base-url
+(defn port-index-protocol
```

Comment:
Nitpick:
```suggestion
(defn port-index->protocol
```

Reply:
I don't like using the `->` syntax in places where it doesn't actually match the required arguments. This would need to be `port-index+service-description->protocol` or something like that to be accurate, but that's way too verbose.

Reply:
Then maybe `retrieve-protocol` ?

Reply:
Renamed.